include/boost/beast/core/impl/flat_buffer.hpp

```diff
@@ -262,17 +262,32 @@ reserve(std::size_t n)
 template<class Allocator>
 void
 basic_flat_buffer<Allocator>::
-shrink_to_fit()
+shrink_to_fit() noexcept
 {
     auto const len = size();
+
     if(len == capacity())
         return;
+
     char* p;
     if(len > 0)
     {
         BOOST_ASSERT(begin_);
         BOOST_ASSERT(in_);
-        p = alloc(len);
+#ifndef BOOST_NO_EXCEPTIONS
+        try
+        {
+#endif
+            p = alloc(len);
```

Comment:
Isn't it an invariant of basic_flat_buffer that size() is less than allocator's max_size()? If so, you don't need to call the checked alloc() function here if you can call allocate() directly.

Reply:
calling `allocate()` directly is just so wordy and cumbersome that I mentally defaulted to calling the member function instead

Reply:
`alloc()` is defined in the same translation unit as its caller. I'd be stunned if the double check isn't elided by the optimiser in all honesty.

Reply:
resolved the comment wording. Noted the redundant test but I am reluctant to change existing logic in this commit. Is it worh opening another issue?

Reply:
...and learning to spell?

Reply:
As I pointed out to Richard `allocator_traits` are not required for `allocate` and `deallocate`, so it's not really all that wordy if you just did: `p = get().allocate(len);` 

(All of the `allocate()` and `deallocate()` in this file can avoid `allocator_traits`).

Reply:
@madmongo1 up to you if you want to make the change that Glen suggests. I wouldn't call it a change in "logic" more like a cosmetic change. Allocators are kind of annoying - Glen, not all of us have memorized all the arcane rules for when you need traits and when you don't.